Implement custom popup for Smartspacer

suphon-t opened the pull request:
## Description

Replaced Smartspacer's popup with Launcher3's popup, and adda shortcut to Smartspacer settings in Smartspace preferences.

<img src="https://github.com/LawnchairLauncher/lawnchair/assets/8080853/6248ee77-1dd3-4271-9614-769d4cad737d" height="500">
<img src="https://github.com/LawnchairLauncher/lawnchair/assets/8080853/8d0808bc-9b41-4a3a-a808-cd4c45d3dfc6" height="500">

## Type of change
<!-- Replace :x: with :white_check_mark: to "check" the specified bullet -->

:white_check_mark: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

suphon-t committed:
Disable minify for debug

suphon-t committed:
Show launcher popup when long pressing smartspacer

suphon-t committed:
Add shortcut to smartspacer settings

KieronQuinn commented:
This looks fine, but it has actually removed functionality. The popup menu dynamically shows more than just customise, it also has options to dismiss and open feedback/about intents (same as Smartspace in the Pixel Launcher). The default popup also implements these, so you can copy that to see what they're meant to do: https://github.com/KieronQuinn/Smartspacer/blob/main/sdk-client/src/main/java/com/kieronquinn/app/smartspacer/sdk/client/views/popup/BalloonPopupFactory.kt

KieronQuinn commented:
> > As you've done the actual work implementing this, I can probably pick it up from here and implement the rest if you'd prefer?
> 
> Yeah, sure. That would be helpful since you know more about the UX. Thanks a lot. Let's do a PR to this branch, and then merge everything together.

Done in #3725

KieronQuinn committed:
Add missing Smartspacer Popup Options + fixes (#3725)

* - Update Smartspacer client SDK to 1.0.4 (contains fixes for popup handling)
- Added options for dismiss, about & feedback, make them all optional and use the provided launchIntent method as it provides a fallback for invalid intents
- Use original Customise option as fallback if provided options are empty

* Reorder items to match Pixel Launcher

suphon-t committed:
Merge branch '13-dev' into smartspacer-popup

Goooler commented on build.gradle:
```diff
@@ -172,8 +172,8 @@ android {
             resValue("string", "derived_app_name", "Lawnchair (Debug)")
-            minifyEnabled true
-            shrinkResources true
+            minifyEnabled false
+            shrinkResources false
```
Why tunning off minify here?
